File: code/library/src/main/java/org/fbme/lib/iec61499/declarations/hierarchies/ResourceHierarchy.java
package org.fbme.lib.iec61499.declarations.hierarchies;

import org.fbme.lib.common.Declaration;
import org.fbme.lib.common.DeclarationPath;
import org.fbme.lib.iec61499.declarations.DeviceDeclaration;
import org.fbme.lib.iec61499.declarations.ResourceDeclaration;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;
import java.util.List;

public final class ResourceHierarchy implements DeclarationPath {

    @NotNull private final DeviceDeclaration myDevice;

    @NotNull private final ResourceDeclaration myResource;

    public ResourceHierarchy(@NotNull DeviceDeclaration device, @NotNull ResourceDeclaration resource) {
        myDevice = device;
        myResource = resource;
    }

    public @NotNull DeviceDeclaration getDevice() {
        return myDevice;
    }

    public @NotNull ResourceDeclaration getResource() {
        return myResource;
    }

    @Override
    public @NotNull List<Declaration> getDeclarations() {
        return Arrays.asList(myDevice, myResource);
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/hierarchies/ResourceHierarchy.kt
package org.fbme.lib.iec61499.declarations.hierarchies

import org.fbme.lib.common.Declaration
import org.fbme.lib.common.DeclarationPath
import org.fbme.lib.iec61499.declarations.DeviceDeclaration
import org.fbme.lib.iec61499.declarations.ResourceDeclaration

class ResourceHierarchy(val device: DeviceDeclaration, val resource: ResourceDeclaration) : DeclarationPath {
    override val declarations: List<Declaration>
        get() = listOf<Declaration>(device, resource)
}
